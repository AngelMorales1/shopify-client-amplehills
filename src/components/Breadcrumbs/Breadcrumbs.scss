@import 'settings';

.Breadcrumbs {
  &__arrow {
    height: auto;
    width: spacing(4);
  }
}
