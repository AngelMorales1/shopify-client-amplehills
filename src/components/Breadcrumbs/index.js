import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import get from 'utils/get';

import { Image } from 'components/base';
import styles from './Breadcrumbs.scss';

const Breadcrumbs = ({ breadcrumbs, className }) => {
  if (!breadcrumbs.length) return null;

  return (
    <div className={cx(className, 'my3 px2 flex items-center text-peach')}>
      <div className="flex pr1 items-center">
        <Image
          className={styles['Breadcrumbs__arrow']}
          src="/assets/images/icon-arrow-left.svg"
        />
      </div>
      {breadcrumbs.map((crumb, i) => {
        const delimiter =
          i < breadcrumbs.length - 1 ? <span className="px1">/</span> : '';

        return (
          <span key={i}>
            <Link className="link-text" to={get(crumb, 'to', '')}>
              <span className="small">{get(crumb, 'label', '')}</span>
            </Link>
            {delimiter}
          </span>
        );
      })}
    </div>
  );
};

Breadcrumbs.propTypes = {
  breadcrumbs: PropTypes.arrayOf(
    PropTypes.shape({
      to: PropTypes.string,
      label: PropTypes.string
    })
  ),
  className: PropTypes.string
};

Breadcrumbs.defaultProps = {
  breadcrumbs: [],
  className: ''
};

export default Breadcrumbs;
