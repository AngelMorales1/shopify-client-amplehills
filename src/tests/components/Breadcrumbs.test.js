import React from 'react';
import { shallow } from 'enzyme';

import Breadcrumbs from 'components/Breadcrumbs';

it('renders without data', () => {
  const component = shallow(<Breadcrumbs />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <Breadcrumbs
      className={'foo'}
      breadcrumbs={[{ label: 'Continue Shopping', to: '/products' }]}
    />
  );

  expect(component).toMatchSnapshot();
});
