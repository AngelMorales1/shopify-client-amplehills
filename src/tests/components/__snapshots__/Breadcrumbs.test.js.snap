// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Breadcrumbs
    breadcrumbs={
        Array [
            Object {
              "label": "Continue Shopping",
              "to": "/products",
            },
          ]
    }
    className="foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex pr1 items-center"
>
          <Image
                    alt=""
                    bg={false}
                    className="w100"
                    onImgLoad={[Function]}
                    src="/assets/images/icon-arrow-left.svg"
                    style={Object {}}
          />
</div>,
        Array [
          <span>
            <Link
                        className="link-text small"
                        replace={false}
                        to="/products"
            >
                        <span>
                                    Continue Shopping
                        </span>
            </Link>
            
</span>,
        ],
      ],
      "className": "foo my3 px2 flex items-center text-peach",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Image
            alt=""
            bg={false}
            className="w100"
            onImgLoad={[Function]}
            src="/assets/images/icon-arrow-left.svg"
            style={Object {}}
/>,
          "className": "flex pr1 items-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "bg": false,
            "children": null,
            "className": "w100",
            "onImgLoad": [Function],
            "src": "/assets/images/icon-arrow-left.svg",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="link-text small"
              replace={false}
              to="/products"
>
              <span>
                            Continue Shopping
              </span>
</Link>,
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Continue Shopping
</span>,
              "className": "link-text small",
              "replace": false,
              "to": "/products",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Continue Shopping",
              },
              "ref": null,
              "rendered": "Continue Shopping",
              "type": "span",
            },
            "type": [Function],
          },
          "",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex pr1 items-center"
>
            <Image
                        alt=""
                        bg={false}
                        className="w100"
                        onImgLoad={[Function]}
                        src="/assets/images/icon-arrow-left.svg"
                        style={Object {}}
            />
</div>,
          Array [
            <span>
              <Link
                            className="link-text small"
                            replace={false}
                            to="/products"
              >
                            <span>
                                          Continue Shopping
                            </span>
              </Link>
              
</span>,
          ],
        ],
        "className": "foo my3 px2 flex items-center text-peach",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Image
              alt=""
              bg={false}
              className="w100"
              onImgLoad={[Function]}
              src="/assets/images/icon-arrow-left.svg"
              style={Object {}}
/>,
            "className": "flex pr1 items-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "",
              "bg": false,
              "children": null,
              "className": "w100",
              "onImgLoad": [Function],
              "src": "/assets/images/icon-arrow-left.svg",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="link-text small"
                replace={false}
                to="/products"
>
                <span>
                                Continue Shopping
                </span>
</Link>,
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Continue Shopping
</span>,
                "className": "link-text small",
                "replace": false,
                "to": "/products",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Continue Shopping",
                },
                "ref": null,
                "rendered": "Continue Shopping",
                "type": "span",
              },
              "type": [Function],
            },
            "",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Breadcrumbs
    breadcrumbs={Array []}
    className=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
